includes/admin/class-wc-rest-payments-orders-controller.php

```diff
@@ -103,7 +117,29 @@ public function register_routes() {
 	 * @param WP_REST_Request $request Full data about the request.
 	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
 	 */
-	public function capture_terminal_payment( $request ) {
+	public function capture_terminal_payment( WP_REST_Request $request ) {
+		return $this->handle_payment_capture( $request, true );
+	}
+
+	/**
+	 * Captures an authorization.
+	 * Use-cases: Merchants manually capturing a payment when they enable "capture later" option.
+	 *
+	 * @param  WP_REST_Request $request Full data about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function capture_authorization( WP_REST_Request $request ) {
+		return $this->handle_payment_capture( $request, false );
+	}
+
+	/**
+	 * Handles payment capture.
+	 *
+	 * @param  WP_REST_Request $request Full data about the request.
+	 * @param  bool            $is_terminal_payment Marks if the request is for a Terminal payment (In Person Payment or Interac at the moment).
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	private function handle_payment_capture( WP_REST_Request $request, bool $is_terminal_payment ) {
```

Comment:
This doesn't look right: `capture_terminal_payment` had a clear focus and goal.
Now it's making even more what is violating the single responsibility.

How can we use the built-in capture functionality from order actions for the new endpoint?

Reply:
> Does this answer your question or am I missing something?

Yes.

> Please let me know what you think when you have a moment.

My main concern remains same: IPP capturing logic (incl. preparing the order for capture) accumulated many edge cases and corrections. IMO the new endpoint should be implemented from scratch and use `\WC_Payment_Gateway_WCPay::capture_charge`.

If hooks allow properly capture payment via admin, the simplest approach for the new endpoint would be have basic validation followed by `\WC_Payment_Gateway_WCPay::capture_charge` call.

Did you try this approach and evaluate if the order meta gets properly populated with this approach?

Reply:
@kalessil turns out that you were right. I reverted the changes to the `capture_terminal_payment` endpoint in [373f9d0](https://github.com/Automattic/woocommerce-payments/pull/4676/commits/373f9d0dabcd4a27363808174c40e2d6677b35e3).

> If hooks allow properly capture payment via admin, the simplest approach for the new endpoint would be have basic validation followed by \WC_Payment_Gateway_WCPay::capture_charge call.

I also followed your suggestion in the `capture_authorization` endpoint. I kept the same validations that the `capture_terminal_payment` uses, but omitted the call to `attach_intent_info_to_order` because is not needed according to my tests. Finally, I do call `mark_payment_capture_completed`, instead of `update_order_status_from_intent` to update the status of the order.